MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1545,6 +1545,15 @@
 			break;
 #endif
 	}
+
+	// iOS and tvOS adjustments necessary when running in the simulator on non-Apple GPUs.
+	// Apple1 used as baseline for detecting Apple Silicon on macOS because Apple7 not supported for tvOS.
+	// Must run after setting native iOS and tvOS values above.
+#if MVK_OS_SIMULATOR
+	if ( !supportsMTLGPUFamily(Apple1) ) {
```

Comment:
I normally use `Apple5`, since that's what the Apple Silicon DTKs had, but I guess that wouldn't be supported on tvOS, even on the simulator... would it?

Reply:
Hmmm. I did some further testing. When running on a simulator, the test for GPU family runs against the device being simulated, not the platform the simulator is running on. So for instance, `Apple5` is not supported on the iPad 7th Gen simulator. Conversely, it IS supported on a different simulator running on macOS without Apple Silicon.

Bottom line, testing for GPU family doesn't test for the platform the simulator is running on.

As a result, I've moved to redefining some of the environment macros in terms of the CPU for which the build is targeted. 

Seems to work now, and be more consistent, but please have a look at the macro logic to see if you can see any issues that I might have missed in the logic path. Or if we are using any of these macros in a manner (rightly or wrongly) that the new logic will break, and which I've not noticed.